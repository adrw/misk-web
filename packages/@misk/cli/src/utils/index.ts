import klaw from "klaw"
import { cd, exec, pwd } from "shelljs"
import * as yargs from "yargs"
import { MiskVersion } from "./changelog"

export * from "./changelog"
export * from "./generate"
export * from "./migrate"

export interface IMiskTabJSON {
  name: string
  output_path: string
  port: number
  relative_path_prefix: string
  slug: string
  tsconfigCompilerOptions: any
  version: MiskVersion
  zipOnBuild: boolean
}

export enum Files {
  "gitignore" = ".gitignore",
  "miskTab" = "miskTab.json",
  "old" = ".old_build_files",
  "package" = "package.json",
  "packageLock" = "package-lock.json",
  "prettier" = "prettier.config.js",
  "tsconfig" = "tsconfig.json",
  "tslint" = "tslint.json",
  "webpack" = "webpack.config.js",
  "yarnLock" = "yarn.lock"
}

export const JsonOptions = { spaces: 2 }

export const cmdHeader = (cmd: string) =>
  console.log(`[${cmd.toUpperCase()}] ${pwd()}`)

export const prebuild = async (...args: any) =>
  execute("miskweb prebuild", ...args)

export const execute = async (cmd: string, ...args: any) => {
  const dir = args[0]
  if (dir) {
    cd(dir)
  }
  pwd().stdout
  const terminal = exec(cmd)
  terminal.stdout
  if (terminal.code) {
    throw new Error(
      `Shell command \`${cmd}\` exited with code ${terminal.code}. ${
        terminal.stderr
      }`
    )
  }
}

export const handleCommand = async (
  args: {
    _: string[]
    e: boolean
    each: boolean
    $0: string
  },
  handlerFn: (...args: any) => void,
  blockedOptions: string[] = []
) => {
  let invalidOptions: string[] = []
  blockedOptions.map((opt: string) => {
    if (opt in args) {
      invalidOptions.push(opt)
    }
  })
  if (invalidOptions.length > 0) {
    console.error(
      `Invalid use of ${invalidOptions.map(
        opt => `-${opt} `
      )} option with command ${args._[0]}.`
    )
    yargs
      .hide(invalidOptions[0])
      .hide("help")
      .hide("version")
      .showHelp()
  } else if (args.each) {
    klaw(".")
      .on("data", (item: any) => {
        if (item.stats.isFile() && item.path.includes("/miskTab.json")) {
          const dir = item.path.split("/miskTab.json")[0]
          cd(dir)
          handlerFn(dir)
        }
      })
      .on("error", (err: Error) => console.error(err))
  } else {
    handlerFn()
  }
}
