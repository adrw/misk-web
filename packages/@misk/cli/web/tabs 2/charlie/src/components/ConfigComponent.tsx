import * as React from "react"
import styled from "styled-components"

const Container = styled.div``

const ConfigEntry = styled.pre`
  font-family: Fira Code, Menlo;
`

const ConfigCode = styled.code`
  font-family: Fira Code, Menlo;
`

export interface IConfigResource {
  name: string
  file: string
}

export interface IConfigProps {
  resources: string
  status: string
}

export default class ConfigComponent extends React.PureComponent<IConfigProps> {
  renderConfig(resource: IConfigResource) {
    return (
      <ConfigEntry>
        <h5>
          <strong>{resource.name}</strong>
        </h5>
        <ConfigCode>{resource.file}</ConfigCode>
      </ConfigEntry>
    )
  }

  render() {
    const { resources, status } = this.props
    return (
      <Container>
        <h1>Config</h1>
        <p>{status}</p>
        {resources &&
          Object.entries(resources).map(([name, file]) =>
            this.renderConfig({ name, file })
          )}
      </Container>
    )
  }
}
