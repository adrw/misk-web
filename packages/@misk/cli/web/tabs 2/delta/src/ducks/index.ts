import {
  dispatchSimpleForm,
  dispatchSimpleNetwork,
  IDispatchSimpleForm,
  IDispatchSimpleNetwork,
  ISimpleFormImmutableState,
  ISimpleFormState,
  ISimpleNetworkImmutableState,
  ISimpleNetworkState,
  SimpleFormReducer,
  SimpleNetworkReducer,
  simpleRootSelector,
  watchSimpleFormSagas,
  watchSimpleNetworkSagas
} from "@misk/simpleredux"
import {
  connectRouter,
  LocationChangeAction,
  RouterState
} from "connected-react-router"
import { History } from "history"
import { AnyAction, combineReducers, Reducer } from "redux"
import { all, AllEffect, fork } from "redux-saga/effects"

/**
 * Redux Store State
 */
export interface IState {
  router: Reducer<RouterState, LocationChangeAction>
  simpleForm: ISimpleFormState
  simpleNetwork: ISimpleNetworkState
}

/**
 * Dispatcher
 */
export interface IDispatchProps
  extends IDispatchSimpleForm,
    IDispatchSimpleNetwork {}

export const rootDispatcher: IDispatchProps = {
  ...dispatchSimpleForm,
  ...dispatchSimpleNetwork
}

/**
 * State Selectors
 */
export const rootSelectors = (state: IState) => ({
  simpleForm: simpleRootSelector<IState, ISimpleFormImmutableState>(
    "simpleForm",
    state
  ),
  simpleNetwork: simpleRootSelector<IState, ISimpleNetworkImmutableState>(
    "simpleNetwork",
    state
  )
})

/**
 * Reducers
 */
export const rootReducer = (history: History): Reducer<any, AnyAction> =>
  combineReducers({
    router: connectRouter(history),
    simpleForm: SimpleFormReducer,
    simpleNetwork: SimpleNetworkReducer
  })

/**
 * Sagas
 */
export function* rootSaga(): IterableIterator<AllEffect> {
  yield all([fork(watchSimpleFormSagas), fork(watchSimpleNetworkSagas)])
}
