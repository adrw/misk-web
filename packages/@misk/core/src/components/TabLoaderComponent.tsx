import * as React from "react"
import Helmet from "react-helmet"
import { IWebTab } from "../utilities"

/**
 * <TabLoaderComponent
 *    tabs={adminDashboardTabs}
 * />
 */

export interface ITabLoaderProps {
  tabs: IWebTab[]
}

const RenderTab = (props: IWebTab) => {
  return (
    <span>
      <Helmet>
        <script async={true} src={`/_tab/${props.slug}/tab_${props.slug}.js`} />
      </Helmet>
      <div id={props.slug} />
    </span>
  )
}

export const TabLoaderComponent = (props: ITabLoaderProps): JSX.Element => {
  if (props.tabs) {
    return (
      <div>
        {props.tabs.map(tab => (
          <RenderTab key={tab.slug} {...tab} />
        ))}
      </div>
    )
  } else {
    return <div />
  }
}
