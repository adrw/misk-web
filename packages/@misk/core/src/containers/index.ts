export * from "./CodePreContainer"
export * from "./DesktopWideOnlyContainer"
export * from "./FlexContainer"
export * from "./MobileNeverContainer"
export * from "./MobileOnlyContainer"
export * from "./ResponsiveContainer"
