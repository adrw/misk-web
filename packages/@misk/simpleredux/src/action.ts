/**
 * Actions
 * string enum of the defined actions that is used as type enforcement for Reducer and Sagas arguments
 */
export enum SIMPLEREDUX {
  // Lifecycle
  SUCCESS = "SIMPLEREDUX_SUCCESS",
  FAILURE = "SIMPLEREDUX_FAILURE",
  MERGE = "SIMPLEREDUX_MERGE",
  // Async HTTP Network Calls
  HTTP_DELETE = "SIMPLEREDUX_HTTP_DELETE",
  HTTP_GET = "SIMPLEREDUX_HTTP_GET",
  HTTP_HEAD = "SIMPLEREDUX_HTTP_HEAD",
  HTTP_PATCH = "SIMPLEREDUX_HTTP_PATCH",
  HTTP_POST = "SIMPLEREDUX_HTTP_POST",
  HTTP_PUT = "SIMPLEREDUX_HTTP_PUT",
  // Redux as UI / Field Input Cache
  CACHE = "SIMPLEREDUX_CACHE", // Merge any value into the state
  CACHE_NUMBER = "SIMPLEREDUX_NUMBER", // Merge a number into the state
  CACHE_TOGGLE = "SIMPLEREDUX_TOGGLE" // Toggle a boolean stored in the state
}
