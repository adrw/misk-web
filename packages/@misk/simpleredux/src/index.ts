export * from "./utilities"
export * from "./simpleForm"
export * from "./simpleNetwork"
// export * from "./simpleRouter" not built yet
