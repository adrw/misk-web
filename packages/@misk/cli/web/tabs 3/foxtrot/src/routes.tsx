import * as React from "react"
import { Route, Switch } from "react-router"
import { LoaderContainer } from "./containers"

const routes = (
  <span>
    <Switch>
      <Route component={LoaderContainer} />
    </Switch>
  </span>
)

export default routes
