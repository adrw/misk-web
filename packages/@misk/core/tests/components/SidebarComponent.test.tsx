import React from "react"
import { cleanup, render } from "react-testing-library"
import { SidebarComponent } from "../../src/components"
import { testTabs } from "../testUtilities"

describe("components", () => {
  afterEach(cleanup)
  it("SidebarComponent can render", () => {
    const { asFragment } = render(<SidebarComponent tabs={testTabs} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
