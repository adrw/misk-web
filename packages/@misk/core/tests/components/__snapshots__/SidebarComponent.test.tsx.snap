// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components SidebarComponent can render 1`] = `
<DocumentFragment>
  .emotion-0 {
  position: absolute;
}

<div
    class="emotion-0"
  >
    <ul
      class="bp3-menu"
    >
      <li
        class=""
      >
        <a
          class="bp3-menu-item bp3-popover-dismiss"
          href="/tab/alpha/"
        >
          <div
            class="bp3-text-overflow-ellipsis bp3-fill"
          >
            Admin :: Tab Alpha
          </div>
        </a>
      </li>
      <li
        class=""
      >
        <a
          class="bp3-menu-item bp3-popover-dismiss"
          href="/tab/bravo/"
        >
          <div
            class="bp3-text-overflow-ellipsis bp3-fill"
          >
            Extra :: Tab Bravo
          </div>
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
